compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4274,19 +4278,26 @@ private boolean equalityIntersectionExists(Set<BType> lhsTypes, Set<BType> rhsTy
         return matchFound;
     }
 
-    boolean validNumericTypeExists(BType type) {
-
+    boolean validNumericStringOrXmlTypeExists(BType type, boolean isNumericType) {
         if (type.isNullable() && type.tag != TypeTags.NIL) {
             type = getSafeType(type, true, false);
         }
-        if (isBasicNumericType(type)) {
+        if (isNumericType) {
+            if (isBasicNumericType(type)) {
+                return true;
+            }
+        } else if (TypeTags.isStringTypeTag(type.tag) || TypeTags.isXMLTypeTag(type.tag)) {
             return true;
         }
+        return validTypeExists(type, isNumericType);
+    }
+
+    boolean validTypeExists(BType type, boolean isNumericType) {
```

Comment:
Can we improve the method name? Not self-explanatory/clear atm.

Reply:
Removed this method.